src/main/java/org/commcare/core/network/ModernHttpRequester.java

```diff
@@ -112,9 +114,30 @@ public void makeRequestAndProcess() {
             default:
                 throw new IllegalArgumentException("Invalid HTTPMethod " + method.toString());
         }
-        return currentCall.execute();
+
+        return executeAndCheckCaptivePortals(currentCall);
+    }
+
+    private Response executeAndCheckCaptivePortals(Call currentCall) throws IOException {
+        try {
+            return currentCall.execute();
+        } catch (SSLHandshakeException | SSLPeerUnverifiedException e) {
+            // This may be a real SSL exception associated with the real endpoint server, or this
+            // might be a property of the local network.
+            if(checkCurrentNetworkAsCaptivePortal()) {
+                throw new CaptivePortalRedirectException(e);
+            }
+
+            //Otherwise just rethrow the original exception. Probably a certificate issue
+            throw e;
+        }
     }
 
+    private boolean checkCurrentNetworkAsCaptivePortal() {
+        return false;
```

Comment:
@ShivamPokhriyal This is the entry point for adding however we want to do detection.

I tested this with some [sample code from the android source](https://stackoverflow.com/a/14030276/5290221) and everything worked, but wasn't sure exactly what mechanism we wanted to rely on here.